openselery/visualization.py

```diff
@@ -11,66 +11,67 @@
-  return float(transaction["native_amount"]["amount"]) < 0 and transaction["native_amount"]["currency"] == "EUR"
+    return float(transaction["native_amount"]["amount"]) < 0 and transaction["native_amount"]["currency"] == "EUR"
 
 def transactionToEur(transaction):
-  assert transaction["native_amount"]["currency"] == "EUR"
-  return float(transaction["native_amount"]["amount"])
+    assert transaction["native_amount"]["currency"] == "EUR"
+    return float(transaction["native_amount"]["amount"])
 
 def drawBarChart(title, xlabel, keys, values):
-  plt.xscale('log')
-  _, diagram = plt.subplots()
-  y_pos = np.arange(len(keys))*4
-  diagram.barh(y_pos, values, align='center', log='true', in_layout='true' )
-  diagram.set_yticks(y_pos)
-  diagram.set_yticklabels(keys)
-  diagram.invert_yaxis()  # labels read top-to-bottom
-  diagram.set_xlabel(xlabel)
-  diagram.set_title(title)
-  diagram.xaxis.set_major_formatter(ScalarFormatter())
-
-def visualizeTransactions(resultFolder):
-  # read transactions file
-  transactions_file = open(resultFolder + '/transactions.txt').read()
-  transactions = json.loads(transactions_file)
-
-  # prepare transaction data
-  data_by_day_last_month = groupBy(filter(lambda t: transactionIsEurSpent(t) and transactionIsLastMonth(t), transactions["data"]), transactionToYearMonthDay)
-  data_by_year_month = groupBy(filter(transactionIsEurSpent, transactions["data"]), transactionToYearMonth)
-  data_by_user = groupBy(filter(transactionIsEurSpent, transactions["data"]), transactionToUserEmail)
-
-  eur_by_day_last_month = { k: -1 * sum(map(transactionToEur, v)) for k,v in data_by_day_last_month.items() }
-  eur_by_year_month = { k: -1 * sum(map(transactionToEur, v)) for k,v in data_by_year_month.items() }
-  eur_by_user = { k: -1 * sum(map(transactionToEur, v)) for k,v in data_by_user.items() }
-
-  # draw diagrams
-  plt.rcdefaults()
-
-  drawBarChart("EUR transactions per day in last month", "EUR", eur_by_day_last_month.keys(), eur_by_day_last_month.values())
-  plt.savefig(resultFolder + "/transactions_per_day.png", bbox_inches = "tight" )
-
-  drawBarChart("EUR transactions per month", "EUR", eur_by_year_month.keys(), eur_by_year_month.values())
-  plt.savefig(resultFolder + "/transactions_per_month.png", bbox_inches = "tight")
-
-  drawBarChart("EUR transactions per user", "EUR", eur_by_user.keys(), eur_by_user.values())
-  plt.savefig(resultFolder + "/transactions_per_user.png", bbox_inches = "tight")
+    plt.xscale('log')
+    _, diagram = plt.subplots()
+    y_pos = np.arange(len(keys))*4
+    diagram.barh(y_pos, values, align='center', log='true', in_layout='true' )
+    diagram.set_yticks(y_pos)
+    diagram.set_yticklabels(keys)
+    diagram.invert_yaxis()  # labels read top-to-bottom
+    diagram.set_xlabel(xlabel)
+    diagram.set_title(title)
+    diagram.xaxis.set_major_formatter(ScalarFormatter())
+
+def visualizeTransactions(resultDir, transactionFilePath):
+    if transactionFilePath:
```

Comment:
Maybe we should check then both arguments.

Reply:
true, but that goes for nearly every parameter for nearly every function used. And that goes even further with proper exceptions and error handling for each function. baby steps :)